debian/control

```diff
@@ -0,0 +1,42 @@
+Source: amazon-ecr-credential-helper
+Section: misc
+Priority: optional
+Maintainer: Samuel Karp <skarp@amazon.com>
+Build-Depends:
+ debhelper (>= 8.0.0),
+ dh-golang (>= 1.1),
+ golang-go (>= 1.7),
+ git,
+ golang-github-aws-aws-sdk-go-dev,
+ golang-github-docker-docker-credential-helpers-dev,
+ golang-github-mitchellh-go-homedir-dev,
+ golang-github-sirupsen-logrus-dev,
+ golang-github-golang-mock-dev,
+ golang-github-pkg-errors-dev
+Standards-Version: 4.3.0
+Homepage: https://github.com/awslabs/amazon-ecr-credential-helper
+Vcs-Browser: https://github.com/awslabs/amazon-ecr-credential-helper
+Vcs-Git: https://github.com/awslabs/amazon-ecr-credential-helper
+XS-Go-Import-Path: github.com/awslabs/amazon-ecr-credential-helper
+
+Package: amazon-ecr-credential-helper
+Architecture: any
+Depends:
```

Comment:
Should this package declare a dependency on docker.io? Is it actually useful without it?

Reply:
I used `Enhances: docker.io` to capture the relationship.  This package does not depend on Docker at runtime, but is designed to work with it.

The Docker project builds packages for Debian as well: https://docs.docker.com/install/linux/docker-ce/debian/.  Those packages are called `docker-ce` instead of `docker.io`.  I'd like this package to work with both the Debian project's `docker.io` package and the Docker project's `docker-ce` package.

Reply:
Enhances and Depends have different meanings. In particular, Enhances is a fairly weak relationship. Depends is much stronger. With Enhances, it's possible to have the credential helper installed with no Docker package at all. Is that a useful configuration? If so, then enhances may be reasonable, though possibly also with a Recommends or Suggests relationship on Docker packages. If not, then Depends is needed.

You can declare a dependency relationship that can be met by the various different docker packages with something like `Depends: docker.io | docker-ce`. `docker.io` should be listed first, to ensure that, if a Docker package is installed in order to fulfill the dependency, it will be the one from the Debian archive.

See [ch 7 of debian-policy](https://www.debian.org/doc/debian-policy/ch-relationships.html) for details, and in particular [section 7.2](https://www.debian.org/doc/debian-policy/ch-relationships.html#binary-dependencies-depends-recommends-suggests-enhances-pre-depends)

Reply:
I really don't think a Depends relationship is warranted here.  The application does not depend on any facility that Docker provides at runtime, and I can conceive of situations where a user might want to install amazon-ecr-credential-helper without Docker (e.g., if they wanted to use amazon-ecr-credential-helper with another Docker registry-compatible container runtime like cri-o, containerd, rkt, or with a utility like [img](https://github.com/genuinetools/img)).

I think Enhances makes sense given that amazon-ecr-credential-helper provides additional usability for Docker, but I think a Recommends or Suggests could be fine as well.

Thanks for telling me about the use of the `|` operator in dependency relationships; I didn't know that was possible.

Should I change Enhances to `Recommends: docker.io | docker-ce`?

Reply:
Per our discussion this morning I've added a `Recommends: docker.io | docker-ce` and changed the Enahnces to `Enhances: docker.io, docker-ce`.